src/custom-elements/gnb/gnb.js

```diff
@@ -167,17 +163,36 @@ export class Gnb extends HTMLElement {
 
     const profileImg = document.createElement("img");
     profileImg.classList.add("profile-img");
-    profileImg.setAttribute("src", this.#prop.profileImgSrc);
+    profileImg.setAttribute("src", this.prop.profileImgSrc);
     profileDiv.append(profileImg);
 
-    const usernameDiv = document.createElement("div");
-    usernameDiv.classList.add("username-container");
-    usernameDiv.textContent = this.#prop.username;
-    myAccountDiv.append(usernameDiv);
+    const emailDiv = document.createElement("div");
+    emailDiv.classList.add("email-container");
+    emailDiv.textContent = this.prop.email;
+    myAccountDiv.append(emailDiv);
 
     this.shadowRoot.querySelector("nav").append(myAccountDiv);
   }
 
+  handleScroll() {
+    let currentScrollPos = window.scrollY;
+    if (this.prevScrollPos > currentScrollPos) {
+      this.documentHeader.style.top = "0";
+      this.documentHeader.classList.add("shadow");
+    } else {
+      this.documentHeader.style.top = "-" + this.navHeight + "px";
+      this.documentHeader.classList.remove("shadow");
+    }
+
+    if (currentScrollPos == 0) {
+      this.documentHeader.classList.remove("shadow");
+    }
+    this.prevScrollPos = currentScrollPos;
```

Comment:
취향의 영역이긴 한데 변수명은 좀 더 구체적으로 적는게 좋다고 생각해요. 
```suggestion
    this.prevScrollPosition = currentScrollPosition;
```